Add docker toolchain support

pagakarthik opened the pull request:
I ran into issues seting up the development environment with jax and other dependencies, and built a docker based environment that supports ksm. ~~I have been testing with the tagged version 0.1.9. The recent QT additions did not play out well :( with this setup. (yet to figure out a work around). But in general this container would help anyone get started quickly with the ksim or ksim-gym.~~ Resolved by installing the qt and it's dependencies, followed by the change to multi-processing context specification by @alik-git and more information [431]( https://github.com/kscalelabs/ksim/issues/431)

Notable features
 - support for running ksim [Or ksim-gym] in nvidia's official jax container with minimal cuda/jax specific installations
 - access to X11 and other gui elements (OpenGL) for the visualization features
- Requirements: docker (with nvidia runtime feature addition), nvidia container toolkit and relevant GPU drivers [no need to install jax etc.]. (might work with podman as well)

Nvidia's JAX image currently support CUDA 12 with latest jax. 
I tried several combinations to find a way to make ksim accessible with CUDA11 and jax [Specificaly for GPU with lower compute capability] but kept running into dependencies issues and backward compatibility issues with JAX. so sad!
 
TODO:
- ~~need to figure out a work around for the QT changes (I haven't used QT before and thus chose to put this in the backlog)~~
- usage a fixed image tag
- update README for getting started
- add note about accessing tensorboard and information about visualizing a policy using `run_mode=view`

I found ksim to be really interesting. Thanks for making it accessible!

pagakarthik committed:
Add docker toolchain support

- support for running ksim with minimal cuda/jax specific installations

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/kscalelabs/ksim?pullRequest=429) <br/>All committers have signed the CLA.

WT-MM commented:
This is super cool! Thanks for making this :)

pagakarthik committed:
Update image to support QtViewer

- add some additional note to the nvidia.driver.resources section
in docker compose allowing users to specify either use the defualt
mode or specify a more curated configuration
- this is particularly suitable for working with egpu(s) and/or a cloud
instance with multiple gpu-devices
- TODO: add information about installing dependencies - nvidia container
 toolkit and the (gpu) driver
- TODO: add information about accessing tensorboard
- TODO: suggest that it might be beneficial to create a ksim-dev
repo with only the docker files and may be include ksim and/or ksim-gym
to develop within the container [as a starter kit] and avoid the
- jax/cuda/ dependencies - installation/ issues

pagakarthik committed:
Merge branch 'kscalelabs:master' into kp/add_docker_env

pagakarthik committed:
Merge branch 'kscalelabs:master' into kp/add_docker_env

pagakarthik committed:
Merge branch 'kscalelabs:master' into kp/add_docker_env

WT-MM committed:
Merge branch 'master' into kp/add_docker_env

WT-MM commented:
Is this ready to test?

codekansas committed:
Merge branch 'master' into kp/add_docker_env
